### PR TITLE
add support for scanning methods on polymer elements and mixins

### PR DESCRIPTION
**Pull request by FredKSchott**
Note that this doesn't change the data returned by analysis yet. That has been addressed separately in the next PR. CHANGELOG will be updated then since there's no visible change to analysis results.

/cc @justinfagnani @rictic @usergenic

### CODE REVIEW

File: src/polymer/polymer-element-scanner.ts [hunk 1]
```diff
@@ -217,9 +233,15 @@ class ElementVisitor implements Visitor {
             getters[scannedPolymerProperty.name] = scannedPolymerProperty;
           } else if (scannedPolymerProperty.setter) {
             setters[scannedPolymerProperty.name] = scannedPolymerProperty;
+          } else if (
+              prop.method === true ||
+              prop.value.type === 'ArrowFunctionExpression' ||
+              prop.value.type === 'FunctionExpression') {
```
Comment (FredKSchott) [inline src/polymer/polymer-element-scanner.ts]: @justinfagnani is there a better way to do this?

File: src/model/method.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Property, ScannedProperty} from './model';
+
+export interface ScannedMethod extends ScannedProperty {
+  function: {
```
Comment (justinfagnani) [inline src/model/method.ts]: why have this nested inside of a function property? Wouldn't methods just have parameters and a return value?

File: src/model/property.ts [hunk 3]
```diff
@@ -21,14 +20,14 @@ import {ScannedFeature} from './model';
...
   type?: string;
-  description?: string;
-  jsdoc?: jsdoc.Annotation;
   private?: boolean;
   'default'?: string;
   readOnly?: boolean;
-  sourceRange: SourceRange|undefined;
-  astNode: estree.Node|null;
   changeEvent?: string;
+  function?: {
```
Comment (justinfagnani) [inline src/model/property.ts]: since you added ScannedMethod, what's this for here?

File: src/polymer/polymer-element-scanner.ts (hunk 1 above)
Comment (justinfagnani) [inline src/polymer/polymer-element-scanner.ts]: Not that I know of

**Pull request merged**